connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -184,7 +184,31 @@ public void resumeConnector(String connector) {
         configBackingStore.putTargetState(connector, TargetState.STARTED);
     }
 
-    @Override
+@Override
+public Map<String, String> maskCredentials(String connName, Map<String, String> config) {
+    Connector connector = getConnector(config.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG));
+    if (connector == null) {
+        throw new NotFoundException("No status found for connector " + connName);
+    }
+
+    ConfigDef configDef = connector.config();
+
+    Map<String, String> newConfig = new LinkedHashMap<>();
+    for (Map.Entry<String, String> entry : config.entrySet()) {
+        final String key = entry.getKey();
+
+        // If it exists in ConfigDef and it is type password
+        if (configDef.configKeys().containsKey(key) && configDef.configKeys().get(key).type.equals(ConfigDef.Type.PASSWORD)) {
```

Comment:
I'd suggest trying to get the key and then checking for null:

```
ConfigKey configKey = configDef.configKeys().get(key);
if (configKey != null && ConfigDef.Type.PASSWORD.equals(configKey.type)) {
    newConfig.put(key, Password.HIDDEN);
} else {
    newConfig.put(key, entry.getValue());
}
```
This is more readable, and the comment then seems superfluous.